In [1]:
import torch
import numpy as np

# Basic operation

### Judge the type of object

In [5]:
# judge whether an object is a tensor or not.
a = np.array([1, 2, 3, 4])
b = torch.randn(3,3)
print("a is a tensor:%s" % torch.is_tensor(a))
print("b is a tensor:%s" % torch.is_tensor(b))


a is a tensor:False
b is a tensor:True


we can also use function:`torch.is_storage(obj)` to judge whether the object is a python storage object.return bool type(true or false)

### Get the element number of tensor 

`torch.numel(input)->int`: return the element number of tensor:input

In [6]:
a = torch.randn(1, 2, 3, 4, 5)
print("the element number of tensor a is: %d" % torch.numel(a))
b = torch.zeros(4, 4)
print("the element number of tensor b is: %d" % torch.numel(b))

the element number of tensor a is: 120
the element number of tensor b is: 16


calculate method: a is 5 dims tensor, total element number is:5! = 120

### Create Ops

create a inentity matrix:
`torch.eye(n, m=None, out=None)`: return a table(2 dims identity matrix) 

**parameters:** 

* n(int):the number of rows.
* m(int, optional):the number of column,if "None":m==n
* out(Tensor, optional): output tensor    